src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -370,15 +370,17 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
     * @return Date
     */ 
     private static Date dtStartOrEndOfAddrSeasonal(Address__c addr, boolean fStart) {    
+        integer startYear = integer.valueOf(addr.Seasonal_Start_Year__c);
```

Comment:
I didn't modify isSeasonalOverlap because of the comment in line 583 (expand code to see) that says "we don't need to know what year or month we are currently in, but we need to take into account days (since we could be in the same month)." But I didn't dig into the logic in depth. I'll take a look.